cli/cmd/compose/run.go

```diff
@@ -72,17 +73,30 @@ func runRun(ctx context.Context, opts runOptions) error {
 		return err
 	}
 
+	dependencies := []types.ServiceConfig{}
+	originalServices := project.Services
 	containerID, err := progress.Run(ctx, func(ctx context.Context) (string, error) {
-		return c.ComposeService().CreateOneOffContainer(ctx, project, compose.RunOptions{
-			Name:    opts.Name,
-			Command: opts.Command,
-		})
+		for _, service := range originalServices {
+			if service.Name != opts.Name {
+				dependencies = append(dependencies, service)
+			}
+		}
+		project.Services = types.Services(dependencies)
```

Comment:
`append` magic allows to declare `dependencies` variable with type `types.Services`